common/SmokeTests/SmokeTest/SmokeTest.csproj

```diff
@@ -1,33 +1,30 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
-
+<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <OutputType>Exe</OutputType>
     <TargetFrameworks>netcoreapp2.1;net461</TargetFrameworks>
+    <LangVersion>latest</LangVersion>
+    <StartupObject>SmokeTest.Program</StartupObject>
   </PropertyGroup>
 
-  <ItemGroup>
-    <PackageReference Include="Azure.Core" Version="1.0.2-dev.*" />
-    <PackageReference Include="Azure.Identity" Version="1.0.0-dev.*" />
-    <PackageReference Include="Azure.Messaging.EventHubs" Version="5.0.0-dev.*" />
-    <PackageReference Include="Azure.Messaging.EventHubs.Processor" Version="5.0.1-dev.*" />
-    <PackageReference Include="Azure.Security.Keyvault.Secrets" Version="4.0.0-dev.*" />
-    <PackageReference Include="Azure.Storage.Blobs" Version="12.0.0-dev.*" />
-    <PackageReference Include="Microsoft.Azure.DocumentDB.Core" Version="2.10.0" />
+  <ItemGroup Condition="'$(TargetFramework)' == 'net461'">
+    <Reference Include="System" />
+    <Reference Include="Microsoft.CSharp" />
   </ItemGroup>
 
   <ItemGroup>
-    <Compile Update="Properties\Resources.Designer.cs">
-      <DesignTime>True</DesignTime>
-      <AutoGen>True</AutoGen>
-      <DependentUpon>Resources.resx</DependentUpon>
-    </Compile>
+    <PackageReference Include="Azure.Core" Version="1.3.0-dev.*" />
+    <PackageReference Include="Azure.Identity" Version="1.2.0-dev.*" />
+    <PackageReference Include="Azure.Messaging.EventHubs" Version="5.1.0-dev.*" />
+    <PackageReference Include="Azure.Messaging.EventHubs.Processor" Version="5.1.0-dev.*" />
+    <PackageReference Include="Azure.Security.Keyvault.Secrets" Version="4.1.0-dev.*" />
+    <PackageReference Include="Azure.Storage.Blobs" Version="12.5.0-dev.*" />
+    <PackageReference Include="Microsoft.Azure.Amqp" Version="2.4.3" />
+    <PackageReference Include="Microsoft.Azure.Devices" Version="1.20.1" />
+    <PackageReference Include="Microsoft.Azure.DocumentDB.Core" Version="2.10.0" />
   </ItemGroup>
 
+  <!-- Sample: IoT Hub Connection String Translation -->
   <ItemGroup>
-    <EmbeddedResource Update="Properties\Resources.resx">
-      <Generator>ResXFileCodeGenerator</Generator>
-      <LastGenOutput>Resources.Designer.cs</LastGenOutput>
-    </EmbeddedResource>
+    <Compile Include="$(MSBuildThisFileDirectory)..\..\..\samples\iothub-connect-to-eventhubs\**\*.cs" Link="Samples\IoTHubConnection\SharedSource\%(Filename)%(Extension)" />
```

Comment:
This links the source directly from the sample project and should now pick up any changes automatically.    All of the sample code is now embedded as part of the main smoke tests library rather than using an external project to keep it segmented.   Instead, I went with a folder structure and some conventions to try and make clear what is directly part of the smoke tests versus shared source from the sample.

**Note:** One of the side effects here is that any changes to the code for the sample in the smoke tests context will be reflected back to the sample directly.   It shouldn't cause issues in the case of doing targeted mutation for a test run, but could be a source of local errors.

Reply:
We should consider just taking a project reference to this library instead of including the source directly. Including the source directly means we will always have to minor any project changes from the samples into the smoketest project which isn't ideal as we scale this out to other samples.

Reply:
I'll leave that decision to @danieljurek.   If using a true project reference, you're likely to end up with version conflicts in the dependencies, since the smoke tests mutate the project to always pull the latest from the daily builds and the referenced sample projects would be using published GA packages.

As a result, Daniel's script to mutate the project would have to be updated to spider each project referenced by the smoke tests, do the daily build transform, and then follow each package referenced by that reference and so on.    _(I would expect that you'd probably not see more than 1-2 levels, given that most assets are using package references)_

I'm certainly open to that, but would prefer that we not try to take on that additional scope in this set of changes.